x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -148,8 +137,8 @@ export const CasesProvider: React.FC<{ value: CasesContextProps }> = ({
     [getFilesClient, owner]
   );
 
-  return isCasesContextValue(value) ? (
```

Comment:
> In the end, what needed to be changed to remove this check?

Some changes in the PR take into account undefined `appId`, mostly in navigation and when we construct the local storage keys. In practice, the `appId` will never be `undefined` when the cases app is rendered.

> I hadn't realized before that isCasesContextValue also checked for value.permissions != null

The `permissions` is coming as a prop and it is required. I am not sure why we had the check but it will never be `undefined`.